### PR TITLE
Don't build anything on a mac

### PR DESCRIPTION
**Pull request by kbrock**
I'm thinking this will work for us

### CODE REVIEW

File: ext/surro-gate/extconf.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 require 'mkmf'
 
 have_header 'sys/epoll.h'
-create_makefile 'surro-gate/selector_ext'
+if RUBY_PLATFORM =~ /linux/ && !defined?(JRUBY_VERSION) && !ENV['SURRO_GATE_NOEXT']
+  create_makefile 'surro-gate/selector_ext'
+else
+  IO.write("Makefile", "all:\n")
```
Comment (skateman) [inline ext/surro-gate/extconf.rb]: https://ruby-doc.org/stdlib-2.5.1/libdoc/mkmf/rdoc/MakeMakefile.html#method-i-dummy_makefile

Comment (kbrock) [reply ext/surro-gate/extconf.rb]: took a while to figure out how to use that method

**Comment by kbrock** [comment]
updated to use `dummy_makefile`
updated to write to current directory

**Comment by skateman** [comment]
I pushed these changes into a `1.0.4.pre.alpha2` gem. Tested this on FreeBSD, but I'd be more comfortable with merging if someone would test it on OSX. Just make a `Gemfile` with this content:
```ruby
source 'https://rubygems.org'
gem 'surro-gate', '=1.0.4.pre.alpha2'
```
Then try to run `bundle install`.

@kbrock @lpichler ?

**Comment by lpichler** [comment]
@skateman 

```
➜  manageiq git:(master) ✗ bundle install
Fetching gem metadata from https://rubygems.org/....
Resolving dependencies...
Using concurrent-ruby 1.1.3
Using bundler 1.17.1
Fetching surro-gate 1.0.4.pre.alpha2 (was 1.0.3)
Installing surro-gate 1.0.4.pre.alpha2 (was 1.0.3) with native extensions
Bundle complete! 1 Gemfile dependency, 3 gems now installed.
Gems in the groups metric_fu and qpid were not installed.
Bundled gems are installed into `./.bundle`
```

**Comment by djberg96** [comment]
@kbrock nice one, don't think I knew about `dummy_makefile`

**Comment by kbrock** [comment]
@djberg96 just learned about it. my first pass had a hardcoded one and skateman said - use this really cool method I just found. (the internet barely knew about it)

**Pull request merged**